### PR TITLE
5.2 - LiveData and LiveData observers

### PR DESCRIPTION
**Pull request by AlexisN94**
- adds encapsulated LiveData to ViewModel of GameFragment
- adds encapsulated LiveData to ViewModel of ScoreFragment

### CODE REVIEW

File: GuessTheWord-Starter/app/src/main/java/com/example/android/guesstheword/screens/game/GameViewModel.kt [hunk 1]
```diff
@@ -1,20 +1,36 @@
...
-import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 
 class GameViewModel : ViewModel() {
 
-    // The current word
-    var word: String = ""
+    // The current _word
```
Comment (andre-exd) [inline GuessTheWord-Starter/app/src/main/java/com/example/android/guesstheword/screens/game/GameViewModel.kt]: I'm guessing this occurred after using a replace all. The previous comment with word seemed correct. Can you please revert? Try to apply to other comments

File: GuessTheWord-Starter/app/src/main/java/com/example/android/guesstheword/screens/game/GameFragment.kt [hunk 2]
```diff
@@ -49,46 +50,42 @@ class GameFragment : Fragment() {
...
+        viewModel.score.observe(viewLifecycleOwner, { newScore ->
+            binding.scoreText.text = newScore.toString()
+        })
+
+        viewModel.word.observe(viewLifecycleOwner, { newWord ->
+            binding.wordText.text = newWord
+        })
+
+        viewModel.eventGameFinish.observe(viewLifecycleOwner, { hasFinished ->
+            if(hasFinished) gameFinished()
```
Comment (andre-exd) [inline GuessTheWord-Starter/app/src/main/java/com/example/android/guesstheword/screens/game/GameFragment.kt]: Fix format of file please

File: GuessTheWord-Starter/app/src/main/java/com/example/android/guesstheword/screens/score/ScoreViewModel.kt [hunk 3]
```diff
@@ -1,7 +1,27 @@
 package com.example.android.guesstheword.screens.score
 
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 
 class ScoreViewModel(finalScore: Int) : ViewModel() {
-    var score = finalScore
+    private val _score = MutableLiveData<Int>()
+private val _eventPlayAgain = MutableLiveData<Boolean>()
```
Comment (andre-exd) [inline GuessTheWord-Starter/app/src/main/java/com/example/android/guesstheword/screens/score/ScoreViewModel.kt]: Indent code properly

**Review by andre-exd — CHANGES_REQUESTED** [review]

File: GuessTheWord-Starter/app/src/main/java/com/example/android/guesstheword/screens/game/GameFragment.kt (hunk 2 above)
Comment (AlexisN94) [reply GuessTheWord-Starter/app/src/main/java/com/example/android/guesstheword/screens/game/GameFragment.kt]: Edit: Done

File: GuessTheWord-Starter/app/src/main/java/com/example/android/guesstheword/screens/game/GameViewModel.kt (hunk 1 above)
Comment (AlexisN94) [reply GuessTheWord-Starter/app/src/main/java/com/example/android/guesstheword/screens/game/GameViewModel.kt]: Done. Please review the changes

File: GuessTheWord-Starter/app/src/main/java/com/example/android/guesstheword/screens/score/ScoreViewModel.kt (hunk 3 above)
Comment (AlexisN94) [reply GuessTheWord-Starter/app/src/main/java/com/example/android/guesstheword/screens/score/ScoreViewModel.kt]: Done. Please review the changes

**Review by andre-exd — CHANGES_REQUESTED** [review]
Please solve conflicts

**Review by andre-exd — APPROVED** [review]

**Pull request merged**